App crashes if there is not network connection.

**Observed behavior**
When starting the app if there is not wifi connection the app crashes

**Desired behavior**
The app should not crash and just mention that the ROS master is not reachable or something similar.

Some additional logcat output:
```
FATAL EXCEPTION: AsyncTask #1
E/AndroidRuntime(21386): Process: eu.intermodalics.tango_ros_streamer, PID: 21386
E/AndroidRuntime(21386): java.lang.RuntimeException: An error occured while executing doInBackground()
E/AndroidRuntime(21386): 	at android.os.AsyncTask$3.done(AsyncTask.java:300)
E/AndroidRuntime(21386): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
E/AndroidRuntime(21386): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
E/AndroidRuntime(21386): 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E/AndroidRuntime(21386): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
E/AndroidRuntime(21386): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(21386): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(21386): 	at java.lang.Thread.run(Thread.java:811)
E/AndroidRuntime(21386): Caused by: org.ros.exception.RosRuntimeException: No non-loopback interface found.
E/AndroidRuntime(21386): 	at org.ros.address.InetAddressFactory.filterInetAddresses(InetAddressFactory.java:74)
E/AndroidRuntime(21386): 	at org.ros.address.InetAddressFactory.newNonLoopback(InetAddressFactory.java:78)
E/AndroidRuntime(21386): 	at eu.intermodalics.tango_ros_streamer.RunningActivity.init(RunningActivity.java:302)
E/AndroidRuntime(21386): 	at eu.intermodalics.tango_ros_streamer.RunningActivity$6.doInBackground(RunningActivity.java:375)
E/AndroidRuntime(21386): 	at eu.intermodalics.tango_ros_streamer.RunningActivity$6.doInBackground(RunningActivity.java:372)
E/AndroidRuntime(21386): 	at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/AndroidRuntime(21386): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/AndroidRuntime(21386): 	... 4 more
W/ActivityManager(  908):   Force finishing activity eu.intermodalics.tango_ros_streamer/.RunningActivity
```

--- a/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
+++ b/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
@@ -300,49 +300,49 @@ public class RunningActivity extends AppCompatRosActivity implements TangoRosNod
 
     @Override
     protected void init(NodeMainExecutor nodeMainExecutor) {
+        NodeConfiguration nodeConfiguration;
         try {
-            NodeConfiguration nodeConfiguration = NodeConfiguration.newPublic(InetAddressFactory.newNonLoopback().getHostAddress());
+            nodeConfiguration = NodeConfiguration.newPublic(InetAddressFactory.newNonLoopback().getHostAddress());
             nodeConfiguration.setMasterUri(this.nodeMainExecutorService.getMasterUri());
-
-            // Create parameter synchronization node to be up-to-date with Dynamic Reconfigure.
-            String[] dynamicParams = {
-                    getString(R.string.publish_device_pose_key),
-                    getString(R.string.publish_point_cloud_key),
-                    getString(R.string.publish_color_camera_key),
-                    getString(R.string.publish_fisheye_camera_key)};
-            // Tango configuration parameters are non-runtime settings for now.
-            // The reason is that changing a Tango configuration parameter requires to disconnect and
-            // reconnect to the Tango service at runtime.
-            String[] tangoConfigurationParameters = {getString(R.string.pref_drift_correction_key)};
-            mParameterNode = new ParameterNode(this, dynamicParams, tangoConfigurationParameters);
-            nodeConfiguration.setNodeName(mParameterNode.getDefaultNodeName());
-            nodeMainExecutor.execute(mParameterNode, nodeConfiguration);
-
-            // Create and start Tango ROS Node
-            nodeConfiguration.setNodeName(TangoRosNode.NODE_NAME);
-            if (TangoInitializationHelper.loadTangoSharedLibrary() !=
-                    TangoInitializationHelper.ARCH_ERROR &&
-                    TangoInitializationHelper.loadTangoRosNodeSharedLibrary()
-                            != TangoInitializationHelper.ARCH_ERROR) {
-                mTangoRosNode = new TangoRosNode();
-                mTangoRosNode.attachCallbackListener(this);
-                TangoInitializationHelper.bindTangoService(this, mTangoServiceConnection);
-                if (TangoInitializationHelper.checkTangoVersionOk(this)) {
-                    nodeMainExecutor.execute(mTangoRosNode, nodeConfiguration);
-                    updateRosStatus(RosStatus.NODE_RUNNING);
-                } else {
-                    updateTangoStatus(TangoStatus.VERSION_NOT_SUPPORTED);
-                    Log.e(TAG, getResources().getString(R.string.tango_version_error));
-                    displayToastMessage(R.string.tango_version_error);
-                }
-            } else {
-                Log.e(TAG, getString(R.string.tango_lib_error));
-                displayToastMessage(R.string.tango_lib_error);
-            }
         } catch (RosRuntimeException e) {
             Log.e(TAG, getString(R.string.network_error));
             displayToastMessage(R.string.network_error);
             return;
+        }
+        // Create parameter synchronization node to be up-to-date with Dynamic Reconfigure.
+        String[] dynamicParams = {
+                getString(R.string.publish_device_pose_key),
+                getString(R.string.publish_point_cloud_key),
+                getString(R.string.publish_color_camera_key),
+                getString(R.string.publish_fisheye_camera_key)};
+        // Tango configuration parameters are non-runtime settings for now.
+        // The reason is that changing a Tango configuration parameter requires to disconnect and
+        // reconnect to the Tango service at runtime.
+        String[] tangoConfigurationParameters = {getString(R.string.pref_drift_correction_key)};
+        mParameterNode = new ParameterNode(this, dynamicParams, tangoConfigurationParameters);
+        nodeConfiguration.setNodeName(mParameterNode.getDefaultNodeName());
+        nodeMainExecutor.execute(mParameterNode, nodeConfiguration);
+
+        // Create and start Tango ROS Node
+        nodeConfiguration.setNodeName(TangoRosNode.NODE_NAME);
+        if (TangoInitializationHelper.loadTangoSharedLibrary() !=
+                TangoInitializationHelper.ARCH_ERROR &&
+                TangoInitializationHelper.loadTangoRosNodeSharedLibrary()
+                        != TangoInitializationHelper.ARCH_ERROR) {
+            mTangoRosNode = new TangoRosNode();
+            mTangoRosNode.attachCallbackListener(this);
+            TangoInitializationHelper.bindTangoService(this, mTangoServiceConnection);
+            if (TangoInitializationHelper.checkTangoVersionOk(this)) {
+                nodeMainExecutor.execute(mTangoRosNode, nodeConfiguration);
+                updateRosStatus(RosStatus.NODE_RUNNING);
+            } else {
+                updateTangoStatus(TangoStatus.VERSION_NOT_SUPPORTED);
+                Log.e(TAG, getResources().getString(R.string.tango_version_error));
+                displayToastMessage(R.string.tango_version_error);
+            }
+        } else {
+            Log.e(TAG, getString(R.string.tango_lib_error));
+            displayToastMessage(R.string.tango_lib_error);
         }
     }
 

--- a/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
+++ b/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
@@ -40,9 +40,12 @@ import android.widget.TextView;
 import android.widget.Toast;
 
 import org.ros.address.InetAddressFactory;
+import org.ros.exception.RosException;
+import org.ros.exception.RosRuntimeException;
 import org.ros.node.NodeConfiguration;
 import org.ros.node.NodeMainExecutor;
 
+import java.net.InetAddress;
 import java.net.URI;
 
 import eu.intermodalics.tango_ros_node.TangoInitializationHelper;
@@ -299,43 +302,49 @@ public class RunningActivity extends AppCompatRosActivity implements TangoRosNod
 
     @Override
     protected void init(NodeMainExecutor nodeMainExecutor) {
-        NodeConfiguration nodeConfiguration = NodeConfiguration.newPublic(InetAddressFactory.newNonLoopback().getHostAddress());
-        nodeConfiguration.setMasterUri(this.nodeMainExecutorService.getMasterUri());
+        try {
+            NodeConfiguration nodeConfiguration = NodeConfiguration.newPublic(InetAddressFactory.newNonLoopback().getHostAddress());
+            nodeConfiguration.setMasterUri(this.nodeMainExecutorService.getMasterUri());
 
-        // Create parameter synchronization node to be up-to-date with Dynamic Reconfigure.
-        String[] dynamicParams = {
-                getString(R.string.publish_device_pose_key),
-                getString(R.string.publish_point_cloud_key),
-                getString(R.string.publish_color_camera_key),
-                getString(R.string.publish_fisheye_camera_key)};
-        // Tango configuration parameters are non-runtime settings for now.
-        // The reason is that changing a Tango configuration parameter requires to disconnect and
-        // reconnect to the Tango service at runtime.
-        String[] tangoConfigurationParameters = {getString(R.string.pref_drift_correction_key)};
-        mParameterNode = new ParameterNode(this, dynamicParams, tangoConfigurationParameters);
-        nodeConfiguration.setNodeName(mParameterNode.getDefaultNodeName());
-        nodeMainExecutor.execute(mParameterNode, nodeConfiguration);
+            // Create parameter synchronization node to be up-to-date with Dynamic Reconfigure.
+            String[] dynamicParams = {
+                    getString(R.string.publish_device_pose_key),
+                    getString(R.string.publish_point_cloud_key),
+                    getString(R.string.publish_color_camera_key),
+                    getString(R.string.publish_fisheye_camera_key)};
+            // Tango configuration parameters are non-runtime settings for now.
+            // The reason is that changing a Tango configuration parameter requires to disconnect and
+            // reconnect to the Tango service at runtime.
+            String[] tangoConfigurationParameters = {getString(R.string.pref_drift_correction_key)};
+            mParameterNode = new ParameterNode(this, dynamicParams, tangoConfigurationParameters);
+            nodeConfiguration.setNodeName(mParameterNode.getDefaultNodeName());
+            nodeMainExecutor.execute(mParameterNode, nodeConfiguration);
 
-        // Create and start Tango ROS Node
-        nodeConfiguration.setNodeName(TangoRosNode.NODE_NAME);
-        if (TangoInitializationHelper.loadTangoSharedLibrary() !=
-                TangoInitializationHelper.ARCH_ERROR &&
-                TangoInitializationHelper.loadTangoRosNodeSharedLibrary()
-                        != TangoInitializationHelper.ARCH_ERROR) {
-            mTangoRosNode = new TangoRosNode();
-            mTangoRosNode.attachCallbackListener(this);
-            TangoInitializationHelper.bindTangoService(this, mTangoServiceConnection);
-            if (TangoInitializationHelper.checkTangoVersionOk(this)) {
-                nodeMainExecutor.execute(mTangoRosNode, nodeConfiguration);
-                updateRosStatus(RosStatus.NODE_RUNNING);
+            // Create and start Tango ROS Node
+            nodeConfiguration.setNodeName(TangoRosNode.NODE_NAME);
+            if (TangoInitializationHelper.loadTangoSharedLibrary() !=
+                    TangoInitializationHelper.ARCH_ERROR &&
+                    TangoInitializationHelper.loadTangoRosNodeSharedLibrary()
+                            != TangoInitializationHelper.ARCH_ERROR) {
+                mTangoRosNode = new TangoRosNode();
+                mTangoRosNode.attachCallbackListener(this);
+                TangoInitializationHelper.bindTangoService(this, mTangoServiceConnection);
+                if (TangoInitializationHelper.checkTangoVersionOk(this)) {
+                    nodeMainExecutor.execute(mTangoRosNode, nodeConfiguration);
+                    updateRosStatus(RosStatus.NODE_RUNNING);
+                } else {
+                    updateTangoStatus(TangoStatus.VERSION_NOT_SUPPORTED);
+                    Log.e(TAG, getResources().getString(R.string.tango_version_error));
+                    displayToastMessage(R.string.tango_version_error);
+                }
             } else {
-                updateTangoStatus(TangoStatus.VERSION_NOT_SUPPORTED);
-                Log.e(TAG, getResources().getString(R.string.tango_version_error));
-                displayToastMessage(R.string.tango_version_error);
+                Log.e(TAG, getString(R.string.tango_lib_error));
+                displayToastMessage(R.string.tango_lib_error);
             }
-        } else {
-            Log.e(TAG, getString(R.string.tango_lib_error));
-            displayToastMessage(R.string.tango_lib_error);
+        } catch (RosRuntimeException e) {
+            Log.e(TAG, getString(R.string.network_error));
+            displayToastMessage(R.string.network_error);
+            return;
         }
     }
 

--- a/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
+++ b/TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java
@@ -40,12 +40,10 @@ import android.widget.TextView;
 import android.widget.Toast;
 
 import org.ros.address.InetAddressFactory;
-import org.ros.exception.RosException;
 import org.ros.exception.RosRuntimeException;
 import org.ros.node.NodeConfiguration;
 import org.ros.node.NodeMainExecutor;
 
-import java.net.InetAddress;
 import java.net.URI;
 
 import eu.intermodalics.tango_ros_node.TangoInitializationHelper;

--- a/TangoRosStreamer/app/src/main/res/values/strings.xml
+++ b/TangoRosStreamer/app/src/main/res/values/strings.xml
@@ -23,6 +23,7 @@
 
     <!-- Strings related to error messages -->
 
+    <string name="network_error">No network found! Please check your network configuration and restart the application.</string>
     <string name="ros_init_error">Unable to init ROS! Do you have a roscore running?</string>
     <string name="tango_lib_error">Unable to load Tango library</string>
     <string name="tango_bind_error">Unable to bind to Tango Service</string>

